### DISCUSSION TITLE
How to access the NSView, NSWindow or CALayers of an X11 or XCB or XID?

### DISCUSSION THREAD

**Discussion by DuncanHopkinsFoundry** [XQuartz/XQuartz#419] [General]
I am using xquartz to handle all the windows handling. But I now need to a macOS Metal surface on one of these windows.
And for this I need access to the System level NS data. Is there any way I can get hold of this data, so I can add the required NSView and CALayers needed to host a Metal rendering surface?

**Comment by jeremyhu** [comment]
Yeah, that's not going to work quite well, unfortunately.  X11 sits on top of libXplugin for window management which sits on top of internal SPIs (similar SPIs that CoreGraphics and AppKit sit on top of).

A better solution here would be to drop the dependency on libXplugin completley and implement COMPOSITE (eg: by finishing libxcwm + xtoq) directly.
